### PR TITLE
Add fake_user field to Profile

### LINKED ISSUE #2131: fake and/or test users on ci and rc generate OAuth errors
This appears to be limited to ci and rc servers and is not a critical issue, other than polluting our logs. 

https://sentry.io/mit-office-of-digital-learning/micromasters/issues/193148559/

```
AttributeError: Due to a lack of support for Client Credentials Grant in edX, you must specify the access token.
  File "dashboard/tasks.py", line 130, in batch_update_user_data_subtasks
    edx_client = EdxApi(user_social.extra_data, settings.EDXORG_BASE_URL)
  File "edx_api/client.py", line 20, in __init__
    "Due to a lack of support for Client Credentials Grant in edX,"

Unable to create an edX client object for student fake.peyton.butler
```

### PR DESCRIPTION
**Pull request by annagav**
#### What are the relevant tickets?
Fix #2131

#### What's this PR do?
Add fake_user field to Profile, set all test profiles that start with 'fake' to `fake_user=True`. Filter out those users when updating cache.

#### How should this be manually tested?
(Required)

### CODE REVIEW

**Comment by giocalitri** [comment]
Can you please fix the failing tests?

File: profiles/migrations/0036_set_fake_user_for_test_profiles.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.10.5 on 2017-05-18 15:59
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+
+def mark_all_fake_profiles(apps, schema_editor):
+    Profile = apps.get_model("profiles", "Profile")
+    Profile.objects.filter(user__username__startswith='fake').update(fake_user=True)
```
Comment (giocalitri) [inline profiles/migrations/0036_set_fake_user_for_test_profiles.py]: maybe you should test for the string `fake.` (just to narrow even more the possibility of false positives.

**Comment by giocalitri** [comment]
please squash before merging 👍

**Comment by noisecapella** [comment]
Sorry I merged the wrong PR :fearful:

**Comment by giocalitri** [comment]
I hope you squashed it 😡

**Comment by noisecapella** [comment]
Yes, I did the squash and merge option

**Pull request merged**